packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -397,6 +401,113 @@ export function TransferPanel() {
+    const l2Network = ethBridger.l2Network
+
+    if (typeof l2Network.nativeToken === 'undefined') {
+      throw new Error('l2 network does not use custom fee token')
+    }
+
+    const customFeeTokenAllowanceForInbox = await fetchErc20Allowance({
+      address: l2Network.nativeToken,
+      provider: l1Provider,
+      owner: walletAddress,
+      spender: l2Network.ethBridge.inbox
+    })
+
+    const amountBigNumber = utils.parseUnits(amount, nativeCurrency.decimals)
+    // We want to bridge a certain amount of the custom fee token, so we have to check if the allowance is enough.
+    if (!customFeeTokenAllowanceForInbox.gte(amountBigNumber)) {
+      const waitForInput = openCustomFeeTokenApprovalDialog()
+      const confirmed = await waitForInput()
+
+      if (!confirmed) {
+        return false
+      }
+
+      const approveCustomFeeTokenTx = await ethBridger.approveFeeToken({
+        l1Signer
+      })
+      await approveCustomFeeTokenTx.wait()
+    }
+
+    return true
+  }
+
+  async function approveCustomFeeTokenForGateway(): Promise<boolean> {
+    if (typeof walletAddress === 'undefined') {
+      throw new Error('walletAddress is undefined')
+    }
+
+    if (!l1Signer) {
+      throw new Error('failed to find signer')
+    }
+
+    if (!selectedToken) {
+      throw new Error('no selected token')
+    }
+
+    const erc20Bridger = await Erc20Bridger.fromProvider(l2Provider)
+    const l2Network = erc20Bridger.l2Network
+
+    if (typeof l2Network.nativeToken === 'undefined') {
+      throw new Error('l2 network does not use custom fee token')
+    }
+
+    const gateway = await erc20Bridger.getL1GatewayAddress(
+      selectedToken.address,
+      l1Provider
+    )
+
+    const customFeeTokenAllowanceForGateway = await fetchErc20Allowance({
+      address: l2Network.nativeToken,
+      provider: l1Provider,
+      owner: walletAddress,
+      spender: gateway
+    })
+
+    // We want to bridge a certain amount of an ERC-20 token, but the retryable fees on the child chain will be paid in the custom fee token.
+    const { retryableData } = await erc20Bridger.getDepositRequest({
+      from: walletAddress,
+      amount: utils.parseUnits(amount, nativeCurrency.decimals),
+      erc20L1Address: selectedToken.address,
+      l1Provider,
+      l2Provider
+    })
+
+    // maxSubmissionCost + gasLimit * maxFeePerGas
+    const retryableFees = retryableData.maxSubmissionCost
+      //
+      .add(retryableData.gasLimit.mul(retryableData.maxFeePerGas))
+
+    // We have to check if the allowance is enough to cover the fees.
+    if (!customFeeTokenAllowanceForGateway.gte(retryableFees)) {
+      const waitForInput = openCustomFeeTokenApprovalDialog()
+      const confirmed = await waitForInput()
+
+      if (!confirmed) {
+        return false
+      }
+
+      const approveCustomFeeTokenTx = await erc20Bridger.approveFeeToken({
+        erc20L1Address: selectedToken.address,
```

Comment:
You don't need to pass in the custom fee token address, as it's a part of the `L2Network` object and the bridger objects have it as a property (https://github.com/OffchainLabs/arbitrum-sdk/pull/310/files#diff-b1894b842df6f4794b6623dc57e9e14c2519fbe5fa5c5dd63403f1185f305cbbR182). 

This flow is for bridging some other ERC-20 token, and we need to pass the address of that token so that we can figure out which gateway to use and direct approvals towards that.

Reply:
ah ok, the param name was confusing